napari/_vispy/vendored/volume.py

```diff
@@ -201,6 +201,106 @@
     return final_color;
 }}
 
+vec4 calculateCategoricalColor(vec4 betterColor, vec3 loc, vec3 step)
+{{   
+    // Calculate color by incorporating lighting
+    vec4 color0 = $sample(u_volumetex, loc);
+    vec4 color1;
+    vec4 color2;
+    float val0 = colorToVal(color0);
+    float val1 = 0;
+    float val2 = 0;
+    int n_dark_axes = 0;
+    
+    // View direction
+    vec3 V = normalize(view_ray);
+    
+    // calculate normal vector from gradient
+    vec3 N; // normal
+    color1 = $sample( u_volumetex, loc+vec3(-step[0],0.0,0.0) );
+    color2 = $sample( u_volumetex, loc+vec3(step[0],0.0,0.0) );
+
+    N[0] = colorToVal(color1) - colorToVal(color2);
+    val1 = colorToVal(color1);
+    val2 = colorToVal(color2);
+    if (val1 < val0) {{
+        if (val2 < val0) {{
+            n_dark_axes = n_dark_axes + int(1);
+        }}
+        
+    }}
+
+    color1 = $sample( u_volumetex, loc+vec3(0.0,-step[1],0.0) );
+    color2 = $sample( u_volumetex, loc+vec3(0.0,step[1],0.0) );
+    N[1] = colorToVal(color1) - colorToVal(color2);
+    
+    val1 = colorToVal(color1);
+    val2 = colorToVal(color2);
+    if (val1 < val0) {{
+        if (val2 < val0) {{
+            n_dark_axes = n_dark_axes + int(1);
+        }}
+        
+    }}
+
+    color1 = $sample( u_volumetex, loc+vec3(0.0,0.0,-step[2]) );
+    color2 = $sample( u_volumetex, loc+vec3(0.0,0.0,step[2]) );
+    N[2] = colorToVal(color1) - colorToVal(color2);
+    
+    val1 = colorToVal(color1);
+    val2 = colorToVal(color2);
+    if (val1 < val0) {{
+        if (val2 < val0) {{
+            n_dark_axes = n_dark_axes + int(1);
+        }}
+        
+    }}
+
+    //betterColor = max(max(color1, color2),betterColor);
```

Comment:
Why is this commented out?

Reply:
Oops - that should be deleted. Leftover from when I copied the function from the "normal" iso shader.

Reply:
fixed